### ISSUE TITLE
Caching problems

### ISSUE THREAD

**Issue by nijel** [django-compressor/django-compressor#838]
My issue looks a bit like #250, but as that is closed for few years, I've rather decided to open new one.

To me it seems that there is some problem in how caches are handled, but this problem does to happen only sometimes (and most likely when the uwsgi process is serving first requests). Unfortunately I haven't yet find way to reproduce this reliably, it just happens sometimes.

In the end I get following two exceptions:

```
  File "/usr/lib/python2.7/dist-packages/compressor/templatetags/compress.py", line 127, in render
    return self.render_compressed(context, self.kind, self.mode, forced=forced)
  File "/usr/lib/python2.7/dist-packages/compressor/templatetags/compress.py", line 99, in render_compressed
    cache_key, cache_content = self.render_cached(compressor, kind, mode)
  File "/usr/lib/python2.7/dist-packages/compressor/templatetags/compress.py", line 79, in render_cached
    cache_content = cache_get(cache_key)
  File "/usr/lib/python2.7/dist-packages/compressor/cache.py", line 136, in cache_get
    val, refresh_time, refreshed = packed_val
TypeError: 'float' object is not iterable
```

```
  File "/usr/lib/python2.7/dist-packages/compressor/filters/css_default.py", line 37, in input
    URL_PATTERN.sub(self.url_converter, self.content))
  File "/usr/lib/python2.7/dist-packages/compressor/filters/css_default.py", line 103, in url_converter
    return self._converter(matchobj, 1, "url(%s%s%s)")
  File "/usr/lib/python2.7/dist-packages/compressor/filters/css_default.py", line 100, in _converter
    return template % (wrap, self.add_suffix(full_url), wrap)
  File "/usr/lib/python2.7/dist-packages/compressor/filters/css_default.py", line 62, in add_suffix
    suffix = get_hashed_mtime(filename)
  File "/usr/lib/python2.7/dist-packages/compressor/cache.py", line 111, in get_hashed_mtime
    mtime = str(int(get_mtime(filename)))
TypeError: int() argument must be a string or a number, not 'tuple'
```

So it seems that code which is expecting to get float gets tuple and vice versa.

I'm using memcached for caching, rest should be pretty much default setup.

I'm on django-compressor 2.1.1, Django 1.10.6 and python 2.7.13.

**Comment by karyon** [comment]
so, `packed_val` comes directly from the cache, retrieved with a cache key created with `get_templatetag_cachekey`, which puts the string `templatetag` into the cache key.

`get_mtime` returns either an mtime from `os.path.getmtime` or from the cache, retrieved with a cache key created with `get_mtime_cachekey`, which puts the string `mtime` into the cache key.

both these cache key creation methods are used in exactly one place each, and each key is used to read or write to the cache in exactly one place, so i really don't see what might go wrong there. on the other hand, it seems unlikely that django or memcached are screwing this up... the only suggestion i have is using another caching backend and see what happens. have you had any success in the meanwhile?

**Comment by nijel** [comment]
I was able to trigger this only once uwsgi is started, but it didn't happen for most of the time, what makes it hard to debug (usually heavy load did help in reproducing). Right now triggering this seems completely impossible for me. Actually it did last happen when doing this report and not since then, before I had it pretty much on every server restart. The only difference on the system seems to be upgrade to Django 1.10.7, but that does contain no related  change...

While looking at the code I came to same conclusion as you, but I just wanted to open report in case somebody else notices something different. Feel free to close this same as #250, as I'm not able to reproduce this anymore.
